### PR TITLE
Added example for cloud STT service

### PR DESCRIPTION
**Pull request by anik-ghosh-au7**
- Added an example to demonstrate the integration of Google Speech-to-Text API with flutter searchbox

- demo -

https://www.loom.com/share/17a49ab1f4fc4299bb705945d2df8903

- walkthrough - 

https://www.loom.com/share/756b0ed69a454561b20d51cd2559976c

### CODE REVIEW

File: flutter_searchbox/example_cloudSTT/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+A basic example to demonstrate the `flutter_searchbox` usage to build search layout with autosuggestions and voice search.
+
+## Getting Started
+
+1. Go to cloud.google.com and create or select a project.
+2. Enable the Speech-to-Text API for that project.
+3. Create a service account.
+4. Download a private key as JSON.
+5. Open audio_converter.dart file and paste the JSON object in here -
```
Comment (bietkul) [inline flutter_searchbox/example_cloudSTT/README.md:11]: @anik-ghosh-au7 Can you add a quick video (without audio) to cover this process?

File: flutter_searchbox/example_cloudSTT/lib/main.dart [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+  @override
+  _HomePageState createState() => _HomePageState();
+}
+
+class _HomePageState extends State<HomePage> {
+  OverlayState overlayState;
+  OverlayEntry overlayEntry;
+
+  setOverlay(bool status, String value, BuildContext context) async {
```
Comment (bietkul) [inline flutter_searchbox/example_cloudSTT/lib/main.dart]: @anik-ghosh-au7 I'm not able to understand what is going here? Please add comments and use names for variables and methods which are self-exploratory.

**Review by bietkul — CHANGES_REQUESTED** [review]
@anik-ghosh-au7 Please follow the following points:
- Code must be minimal, readable, and easy to understand.
- At the top of the class describe the component usage and explain functionality in brief
- Describe complex functionalities in comments
- Use variables, methods, and class names in a way that they are self-exploratory

File: flutter_searchbox/example_cloudSTT/README.md (hunk 1 above)
Comment (anik-ghosh-au7) [reply flutter_searchbox/example_cloudSTT/README.md:11]: @bietkul - https://recordit.co/WOHanl0NRX

File: flutter_searchbox/example_cloudSTT/lib/main.dart [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+                        enablePopularSuggestions: true,
+                        showAutoFill: true,
+                        maxPopularSuggestions: 3,
+                        size: 5,
+                        dataField: [
+                          {'field': 'original_title', 'weight': 1},
+                          {'field': 'original_title.search', 'weight': 3}
+                        ],
+                        customActions: [
+                          // passing the Recorder as a custom action to the SearchBox widget
```
Comment (bietkul) [inline flutter_searchbox/example_cloudSTT/lib/main.dart]:
```suggestion
                          // using the custom action(mic icon) to handle voice search
```

**Comment by anik-ghosh-au7** [comment]
@siddharthlatest 

In order to merge the PR - 
      we have to use the latest flutter sdk and upgrade the versions for the following packages - 
      - path_provider - 2.0.2 (null safety)
      - google_speech - 2.0.1 (null safety)
      
      use an alternative package to get the audio input from the user
      - flutter_audio_recorder (null safety not available)

**Comment by siddharthlatest** [comment]
@anik-ghosh-au7 Here's the library with null safety included in it: https://github.com/devaliakbar/flutter_audio_recorder. We should now be able to create / merge this PR. We can also resolve the conflicts over here.

**Comment by anik-ghosh-au7** [comment]
demo latest - https://loom.com/share/34df5c19521f42c699745e1a08783a78

**Pull request closed without merging**